R/analyse_pseudo_bulk.R

```diff
@@ -0,0 +1,331 @@
+                                    }
+  )
+  
+  
+  subcluster_ref <- subcluster.out[[k_variable[[2]]]]
+  subcluster_comp <-  subcluster.out[[k_variable[[3]]]]
+  
+  aggregated_counts_subcluster_ref <- scuttle::aggregateAcrossCells(subcluster_ref, ids=subcluster_ref$subcluster)
+  aggregated_counts_subcluster_comp <- scuttle::aggregateAcrossCells(subcluster_comp, ids=subcluster_comp$subcluster)
+  
+  assay_data_ref <- as.data.frame(SingleCellExperiment::assay(aggregated_counts_subcluster_ref))
+  assay_data_comp <- as.data.frame(SingleCellExperiment::assay(aggregated_counts_subcluster_comp))
+  
+  re <- cbind(assay_data_ref, assay_data_comp)
+  cols <- as.list(colnames(re))  # change column names
+  cols <- lapply(cols, function(x) gsub("\\.", "_", x))
+  colnames(re) <- cols
+  
+  return (re)
+}
+
+
+
+
+#' split Seurat object by variable 
+#' @param group_by      entry in metadata table, based on these cluster annotation pseudo bulk is performed
+#' @param k_variable    variable dependent on the split_by -> meta data entry
+#'
+#' @returns             returns pseudo bulk generated data
+#' @importFrom Seurat
+split_variable <- function(seurat_object, group_by, k_variable){
+  seurat_object <- Seurat::AggregateExpression(seurat_object, assays = "RNA", return.seurat = T, group.by = c(group_by,k_variable))
+  tail(Seurat::Cells(seurat_object))
+  seurat_df <- as.data.frame(seurat_object@assays$RNA$counts)
+  
+    return(seurat_df) 
+}
+
+
+#' split Seurat object by random pooling 
+#' @param group_by      entry in metadata table, based on these cluster annotation pseudo bulk is performed
+#' @param k_variable    number of random pools
+#'
+#' @returns             returns pseudo bulk generated data
+#' @importFrom Seurat
+split_variable_random <- function(seurat_object, group_by, k_variable){
+  nrow(seurat_object@meta.data)
+  seurat_object$rand_column <- sample(1:k_variable, nrow(seurat_object), replace = TRUE)
+  
+  seurat_object <- Seurat::AggregateExpression(seurat_object, assays = "RNA", return.seurat = T, group.by = c(group_by,"rand_column"))
+  tail(Seurat::Cells(seurat_object))    
+  seurat_df <- as.data.frame(seurat_object@assays$RNA$counts)
+
+  return(seurat_df)
+}
+
+
+#' method implementation subclustering
+#' @param group_by      entry in metadata table, based on these cluster annotation pseudo bulk is performed
+#' @param alg           Seurat subclustering algorithm id
+#' @param cluster1      cluster to subcluster
+#' @param cluster2      cluster to subcluster
+#' @param k_variable    number of random pools
+#'
+#' @returns             returns pseudo bulk generated data
+#' @importFromt Seurat
+split_clustering <- function(seurat_object, group_by, res, alg, cluster1, cluster2){
+  cluster_ids <- list()
+  cluster_ids <- append(cluster_ids, cluster1)
+  cluster_ids <- append(cluster_ids, cluster2)
+  
+  result_dataframe <- data.frame()
+  
+  # setup seurat object 
+  Seurat::Idents(seurat_object) <- group_by     
```

Comment:
Do you really need to change the object's `Idents`?
I believe that this interfers with the user's object - which we shouldn't do

Reply:
Needs to be addressed